x-pack/plugins/monitoring/server/lib/kibana/get_kibanas.ts

```diff
@@ -117,12 +112,12 @@ export async function getKibanas(
   const response: ElasticsearchResponse = await callWithRequest(req, 'search', params);
   const instances = response.hits?.hits ?? [];
 
-  return instances.map((hit) => {
+  return instances.map((hit: ElasticsearchResponseHit) => {
     const legacyStats = hit._source.kibana_stats;
     const mbStats = hit._source.kibana?.stats;
 
     const kibana: Kibana = {
-      kibana: hit._source.kibana?.kibana ?? legacyStats?.kibana,
+      kibana: buildKibanaInfo(hit),
       concurrent_connections:
         mbStats?.concurrent_connections ?? legacyStats?.concurrent_connections,
       process: {
```

Comment:
Maybe out of scope: But for the code below, I wonder if we can write some helper function like `getProperty(propertyName, hit)`, which would hide the legacyDats, mbStats, and all the optional chaining and ?? operators? 
I wonder if that would make it a bit more readable?

Internally I'd imagine we give a definite name to each property we use, and two functions that know how to find that property from a hit, then the function itself just grabs these to selectors, and chains them together (i the first one is null, use the second one) to find the values.

Reply:
That would be a good improvement for sure. similar conditionals/fallbacks are spread everywhere in the codebase so I'd rather handle this as part of a separate refactoring effort :)